src/Microsoft.AspNetCore.Hosting/Internal/RequestServicesContainerMiddleware.cs

```diff
@@ -27,6 +30,18 @@ public RequestServicesContainerMiddleware(RequestDelegate next, IServiceProvider
             _services = services;
             _next = next;
         }
+        
+        public IServiceScopeFactory ScopeFactory
+        {
+            get
+            {
+                if (_scopeFactory == null)
```

Comment:
Didn't bother locking, worst case scenario, we call into the DI container twice (we were doing it every request before).